extensions/script/ExecuteScript.cpp

```diff
@@ -79,80 +79,72 @@ void ExecuteScript::onSchedule(core::ProcessContext *context, core::ProcessSessi
   context->getProperty(ScriptBody.getName(), script_body_);
   context->getProperty(ModuleDirectory.getName(), module_directory_);
 
-  if (script_file_.empty() && script_engine_.empty()) {
+  if (script_file_.empty() && script_body_.empty()) {
     logger_->log_error("Either Script Body or Script File must be defined");
     return;
   }
 }
 
 void ExecuteScript::onTrigger(const std::shared_ptr<core::ProcessContext> &context,
                               const std::shared_ptr<core::ProcessSession> &session) {
-  try {
-    std::shared_ptr<script::ScriptEngine> engine;
+  std::shared_ptr<script::ScriptEngine> engine;
 
-    // Use an existing engine, if one is available
-    if (script_engine_q_.try_dequeue(engine)) {
-      logger_->log_debug("Using available %s script engine instance", script_engine_);
-    } else {
-      logger_->log_info("Creating new %s script instance", script_engine_);
-      logger_->log_info("Approximately %d %s script instances created for this processor",
-                        script_engine_q_.size_approx(),
-                        script_engine_);
+  // Use an existing engine, if one is available
+  if (script_engine_q_.try_dequeue(engine)) {
+    logger_->log_debug("Using available %s script engine instance", script_engine_);
+  } else {
+    logger_->log_info("Creating new %s script instance", script_engine_);
+    logger_->log_info("Approximately %d %s script instances created for this processor",
+                      script_engine_q_.size_approx(),
+                      script_engine_);
 
-      if (script_engine_ == "python") {
+    if (script_engine_ == "python") {
 #ifdef PYTHON_SUPPORT
```

Comment:
I know it's old code, but I feel like these ifdefs are unnecessary here. 
Python being enabled in the build could also be checked in OnSchedule. In case it's not, the processor can simply thrown an exception there, preventing ontrigger to be called ever.